src/video/video.c

```diff
@@ -196,9 +201,29 @@ static inline void set_screen_viewport(void) {
     glViewport(vp_x, vp_y, g_video_state.viewport_w, g_video_state.viewport_h); // This is used for screen shakes.
 }
 
+void video_render_finish_offscreen(void) {
+    object_array_finish(g_video_state.objects);
+
+    // Set to VGA emulation state, and render to an indexed surface
+    glViewport(0, 0, NATIVE_W, NATIVE_H);
+    object_array_batch batch;
+    object_array_begin(g_video_state.objects, &batch);
+    activate_program(g_video_state.palette_prog_id);
+    render_target_activate(g_video_state.target);
+
+    object_array_blend_mode mode;
+    while(object_array_get_batch(g_video_state.objects, &batch, &mode)) {
+        video_set_blend_mode(mode);
+        object_array_draw(g_video_state.objects, &batch);
+    }
+
+    // Disable render target, and dump its contents as RGBA to the screen.
+    render_target_deactivate();
```

Comment:
deactivate should be right before the RGBA phase, since it resets any and all FBO selection.